### ISSUE TITLE
Cannot build stable tag docs for opentelemetry-python

### ISSUE THREAD

**Issue by xrmx** [readthedocs/readthedocs.org#11850]
## Details

* Read the Docs project URL: https://readthedocs.org/projects/opentelemetry-python/
* Build URL (if applicable): https://readthedocs.org/projects/opentelemetry-python/builds/26567150/
* Read the Docs username (if applicable):

## Expected Result

Compiling for `stable` tag works fine 

## Actual Result

I get a build error because rtd checkouts a random sha (14716484698ece6088e2e84701f5b37b7c066db8) instead of the stable tag. I'm not able to find anything related to the admin panel and haven't digged in the code to see where this may come from.
Thanks!

**Comment by humitos** [comment]
I triggered a new build for your `latest` version and then for `stable` and it worked: https://readthedocs.org/projects/opentelemetry-python/builds/26567839/

I think this is a known issue where Read the Docs needs to "re-sync the VCS branches/tags" to update their identifiers. I'm not finding the issue, but I know we have one opened for this case.

**Comment by xrmx** [comment]
> I triggered a new build for your `latest` version and then for `stable` and it worked: https://readthedocs.org/projects/opentelemetry-python/builds/26567839/

Nice!

> I think this is a known issue where Read the Docs needs to "re-sync the VCS branches/tags" to update their identifiers. I'm not finding the issue, but I know we have one opened for this case.

Is "re-sync the VCS branches/tags" something I can trigger somehow?

Thanks @humitos, hope you are doing well!

**Comment by humitos** [comment]
> Is "re-sync the VCS branches/tags" something I can trigger somehow?

Yes. You can trigger a build for any working version and it will automatically re-sync

**Comment by xrmx** [comment]
> > Is "re-sync the VCS branches/tags" something I can trigger somehow?
> 
> Yes. You can trigger a build for any working version and it will automatically re-sync

Compiling a new version manually is what I was doing yesterday and this morning. Still the hash of the commit was an object that did not exist in the repository. 

Looking at the code the interesting bit would be to understand why a sha that does not belong to the repo ended up there:
https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/doc_builder/director.py#L212

**Comment by stsewd** [comment]
This is the issue about using the commit instead of the actual tag name when re-building https://github.com/readthedocs/readthedocs.org/issues/10838

> Looking at the code the interesting bit would be to understand why a sha that does not belong to the repo ended up here:

That usually happens when there is a webhook from another repository pointing to the wrong RTD project.

Closing as this was solved, and we are tracking the problem about stale commits in another issue.
